### PR TITLE
Keeping your forked branch up to date

### PR DESCRIPTION
**Pull request by Ridhwana**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

The question on how to keeping your forked branch up to date was asked on the slack group, and I think that contributors would benefit from this information as well, as it's not super straightforward.

Also, the sync_fork.sh should not bomb out with `fatal: remote upstream already exists` when teh remote is added, it should simply skip that step.

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
Screenshot 2020-03-09 at 12 58 08

## Added tests?

- [ ] yes
- [x] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by Ridhwana** [comment]
> Cool! I use rebase but merge is fine too :)
> 
> We should probably align what we say in `workflow.md` with these new instructions, can you double check we don't direct people in two different directions?
> 
> https://github.com/thepracticaldev/dev.to/blob/master/docs/getting-started/workflow.md#synchronizing-a-fork-with-upstream--integrate-latest-changes
> 
> Thanks a lot for this!

Hmm this is a good point @rhymes. 
So in workflow.md we do the ffg:

```
git remote add upstream https://github.com/thepracticaldev/dev.to
git fetch upstream
git checkout master
git merge upstream/master
```
Also, it seems we can only use this script when we're getting started, when running it the second time we get a `fatal: remote upstream already exists.`.

So based on the above, we use a merge in the last step, but we use a rebase to pull the master branch, so I was thinking of amending like this:

Option 1:
```shell
git checkout master
git pull --rebase upstream master
git checkout <feature-branch-name>
git merge master
git commit -m "Fix merge conflicts"
git push origin <feature-branch-name>
```
OR 

Option 2 whilst in your feature branch:
```shell
git fetch upstream
git merge upstream/master
git commit -m "Fix merge conflicts"
git push origin <feature-branch-name>
```

Thoughts?

**Comment by rhymes** [comment]
I have had my `git pull` aliased to `git pull --rebase` since forever so I vote for 1 :D Don't know which is one is better 🗡 

Maybe we should update `sync_fork.sh` so that it doesn't fail the second time mmm

**Comment by Ridhwana** [comment]
> I have had my `git pull` aliased to `git pull --rebase` since forever so I vote for 1 :D Don't know which is one is better 🗡
> 
> Maybe we should update `sync_fork.sh` so that it doesn't fail the second time mmm

I also prefer one, it seems more natural to me, and Yeah I was thinking that as well! I'll do the update!

File: scripts/sync_fork.sh [hunk 1]
```diff
@@ -1,8 +1,12 @@
 #!/bin/sh
 set -e
-# sync this fork with upstream to get the latest changes
-git remote add upstream https://github.com/thepracticaldev/dev.to
+
+URL="https://github.com/thepracticaldev/dev.to";
+
+if ! git config remote.upstream.url > /dev/null; then
```
Comment (rhymes) [inline scripts/sync_fork.sh]: so neat!

**Pull request merged**